File: BOJ/Algorithms/Greedy/21758.cpp
#include <iostream>
#include <vector>

using namespace std;

int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);
    int N;
    cin >> N;
    vector<int> data(N);
    vector<int> sumofData(N);
    for(int i = 0; i < N; i++){
        cin >> data[i];
        sumofData[i] = data[i];
    }
    for(int i = 1; i < N; i++) sumofData[i] += sumofData[i-1];
    int answer = 0;
    for(int i = 1; i < N-1; i++){
        answer = max(answer, 2*sumofData[N-1]-data[0]-data[i]-sumofData[i]);
    }
    for(int i = 1; i < N-1; i++){
        answer = max(answer, sumofData[N-1]-data[N-1]-data[i]+sumofData[i-1]);
    }
    for(int i = 1; i < N-1; i++){
        answer = max(answer, sumofData[i]-data[0] + sumofData[N-1]-sumofData[i-1]-data[N-1]);
    }
    cout << answer << '\n';
}

File: BOJ/Algorithms/Greedy/21758.py
from copy import deepcopy
import sys ; input = lambda : sys.stdin.readline().rstrip()

N = int(input())
dataList = list(map(int, input().split()))
sumofData = deepcopy(dataList)
answer = 0
for i in range(1, N):
    sumofData[i] += sumofData[i-1]

for i in range(1, N-1):
    answer = max(answer, 2*sumofData[-1]-dataList[0]-dataList[i]-sumofData[i])
for i in range(1, N-1):
    answer = max(answer, sumofData[-1]-dataList[-1]-dataList[i]+sumofData[i-1])
for i in range(1, N-1): 
    answer = max(answer, sumofData[i]-dataList[0] + sumofData[-1]-sumofData[i-1]-dataList[-1])

print(answer)